### PR TITLE
spectral extraction: improved logic for initial position of "Manual" background trace

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request improves the logic for the initial guess for the position of the "Manual" background trace.  Previously it started at the same guess as the extraction trace (based on the brightest flux medianed over the dispersion direction).  Now it uses the same location that would be adopted for a OneSided background, with the "direction" determined on whether the extraction trace is in the top or bottom half of the image, and the separation at 10% of the cross-dispersion "height".

Motivated by @ojustino's comment: https://github.com/spacetelescope/jdaviz/pull/1682#pullrequestreview-1142409128

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by kecnry** [comment]
Good point, I didn't explain the logic very well.  The sign for the "OneSided" case is only applied if the separation would push the "TwoSided" case off the edge of the image, since the `bg_separation` traitlet is shared between "OneSided" and "TwoSided".  In other words, the same _logic_ that is used when "OneSided" becomes the default is used for the "Manual" position now, but `bg_separation` might still default to positive.

We could separate the `bg_separation` between the One and Two-Sided cases, but I personally don't think that is worth the overhead.  The other option would be to have the "Manual" always match the position of the "OneSided" cases, if you think that would be more intuitive?  (So always above the brightest pixel unless that would force it to go above the image, in which case it would be placed below instead).

**Comment by ojustino** [comment]
I agree that a simpler approach is better here and think having the "Manual" position match the "OneSided" position makes sense.

File: jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py [hunk 1]
```diff
@@ -318,20 +318,24 @@ def _trace_dataset_selected(self, msg=None):
         default_bg_width = int(np.ceil(width / 10))
         default_width = min(default_bg_width, distance_from_edge * 2)
 
+        # sign for one-sided and single trace-pixel depending on whether the brightest pixel is
+        # above or below the middle of the image
+        if default_bg_width * 2 >= distance_from_edge:
+            sign = 1 if (brightest_pixel < width / 2) else -1
+            default_bg_separation = sign * default_bg_width * 2
```
Comment (kecnry) [inline jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py:325]: these lines will be covered by #1739 (coverage was lost when skipping the remote test in #1736)

**Review by ojustino — APPROVED** [review]
The new commit sufficiently addresses my previous comment, and understood about the failing test. Should a merge here wait until the test has been rewritten? I imagine this branch would pass, but the change is also not an extremely urgent one.

**Comment by kecnry** [comment]
I think since we know this case works, it's ok to get in with the slightly lower coverage?  This is the exact same case that was covered before, so these lines will be covered once that test is rewritten.  But I'm also ok letting this PR sit as long as that test doesn't fall off the radar too long - I agree this PR isn't urgent and don't expect merge conflicts.

**Review by pllim — APPROVED** [review]
Code looks like code and we have a follow-up issue, so merging. Thanks!

**Pull request merged**